forge-game/src/main/java/forge/game/card/CardState.java

```diff
@@ -748,6 +748,21 @@ public void addIntrinsicKeywords(Collection<KeywordInterface> intrinsicKeywords2
         }
     }
 
+    public void resetOriginalHost() {
+        final List<CardTraitBase> allAbs = ImmutableList.<CardTraitBase>builder()
```

Comment:
When such immutable list is used for multiple functions, we better separate it into its own function